torch/testing/_creation.py

```diff
@@ -13,6 +13,16 @@
                                            torch.complex128: torch.float64}
 float_to_corresponding_complex_type_map = {v: k for k, v in complex_to_corresponding_float_type_map.items()}
 
+
+def _uniform_random(t: torch.Tensor, low: float, high: float):
+    # uniform_ requires to-from <= std::numeric_limits<scalar_t>::max()
+    # Work around this by scaling the range before and after the PRNG
+    if high - low >= torch.finfo(t.dtype).max:
+        return t.uniform_(low / 2, high / 2).mul_(2)
```

Comment:
Should this error if the reduced range is still too large?

Reply:
I don't think it's possible for this to fail. However, if I'm wrong then `uniform_` will error directly.

The reason I don't think it can fail is that `low` and `high` are clamped to between `finfo(dtype).min` and `finfo(dtype).max`, so I think the largest possible value for `high/2 - low/2` is exactly `finfo(dtype).max`.